pkg/trace/agent/obfuscate.go

```diff
@@ -176,3 +176,62 @@ func (cco *ccObfuscator) MetaHook(k, v string) (newval string) {
 	}
 	return v
 }
+
+// MetaStructHook checks the message inside `v` for credit card information and obfuscates it.
+func (cco *ccObfuscator) MetaStructHook(k string, v []byte) (newval []byte) {
+	if k != "appsec" {
+		// Do not obfuscate unknown structures
+		log.Debugf("Obfuscating unknown meta struct is not supported for key: %v", k)
+		return v
+	}
+	var (
+		changed      bool
+		appsecstruct pb.AppSecStruct
+	)
+	_, err := appsecstruct.UnmarshalMsg(v)
+	if err != nil {
+		// Not an appsec struct, ignore the value and log an error
+		log.Errorf("Error obfuscating appsec struct: %v", err)
+		return v
+	}
+	triggers := appsecstruct.GetTriggers()
+	if triggers == nil {
+		return v
+	}
+	for _, trigger := range triggers {
```

Comment:
```suggestion
	for _, trigger := range appsecstruct.GetTriggers() {
```
A `nil` slice can be iterated over, it's no problem. No need to check. The loop simply won't happen when it is `nil`.